torch/_functorch/eager_transforms.py

```diff
@@ -585,9 +607,25 @@ def compute_jacobian_preallocate_and_copy():
         # Step 2: The returned jacobian is one big tensor per input. In this step,
         # we split each Tensor by output.
         flat_jacobians_per_input = [result.split(flat_output_numels, dim=0) for result in flat_jacobians_per_input]
+
+        def split_and_complex_to_real(out_idx, split, out, primal):
+            split = split.view(out.shape + primal.shape)
+
+            # Complex `n` space can be thoughts of as Real `2n` space.
+            if split.dtype.is_complex:
+                # For complex input `split` will be complex.
+                return torch.view_as_real(split.resolve_conj()).mT
```

Comment:
It turns out that this transpose is here because we return the Jacobian in the `jacrev` case, rather than the adjoint of the Jacobina, which is what `jacrev` naturally computes.